File: factors.c
#include <stdio.h>
#include <stdlib.h>

void factorize(long long n) {
    long long i = 2;
    while (i * i <= n) {
        if (n % i == 0) {
            printf("%lld=%lld*%lld\n", n, i, n / i);
            return;
        }
        i++;
    }
    printf("%lld is a prime number.\n", n);
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: %s <input.txt>\n", argv[0]);
        return 1;
    }

    FILE *file = fopen(argv[1], "r");
    if (file == NULL) {
        perror("Error opening the file");
        return 1;
    }

    char buffer[256];
    while (fgets(buffer, sizeof(buffer), file)) {
        long long num = atoll(buffer); // Convert to long long
        factorize(num);
    }

    fclose(file);
    return 0;
}

File: factors.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class factors {
    public static void factorize(long n) {
        long i = 2;
        while (i * i <= n) {
            if (n % i == 0) {
                System.out.println(n + "=" + i + "*" + n / i);
                return;
            }
            i++;
        }
        System.out.println(n + " is a prime number.");
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: java Factorize input.txt");
            System.exit(1);
        }

        String filePath = args[0];
        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
            String line;
            while ((line = reader.readLine()) != null) {
                long num = Long.parseLong(line);
                factorize(num);
            }
        } catch (IOException e) {
            System.err.println("Error reading the file: " + e.getMessage());
            System.exit(1);
        }
    }
}

